gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java

```diff
@@ -251,12 +257,43 @@ private static BlobWriteSessionConfig getSessionConfig(AsyncWriteChannelOptions
             writeOptions.getTemporaryPaths().stream()
                 .map(x -> Paths.get(x))
                 .collect(ImmutableSet.toImmutableSet()));
+      case PARALLEL_COMPOSITE_UPLOAD:
+        ExecutorService pcuThreadPool =
+            Executors.newCachedThreadPool(
+                new ThreadFactoryBuilder()
+                    .setNameFormat("gcsio-storage-client-pcu-pool-%d")
+                    .build());
+        return BlobWriteSessionConfigs.parallelCompositeUpload()
+            .withBufferAllocationStrategy(
+                BufferAllocationStrategy.fixedPool(
+                    writeOptions.getPCUBufferCount(), writeOptions.getPCUBufferCapacity()))
+            .withPartCleanupStrategy(getPartCleanupStrategy(writeOptions.getPartFileCleanupType()))
+            .withExecutorSupplier(ExecutorSupplier.useExecutor(pcuThreadPool))
+            .withPartNamingStrategy(getPartNamingStrategy(writeOptions.getPartFilePrefix()));
       default:
         return BlobWriteSessionConfigs.getDefault()
             .withChunkSize(writeOptions.getUploadChunkSize());
     }
   }
 
+  private static PartCleanupStrategy getPartCleanupStrategy(PartFileCleanupType cleanupType) {
+    switch (cleanupType) {
+      case NEVER:
+        return PartCleanupStrategy.never();
+      case ON_SUCCESS:
+        return PartCleanupStrategy.onlyOnSuccess();
+      default:
```

Comment:
let us explicity add Always case and throw error in the default case?

Reply:
updated.